ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/resources/File.java

```diff
@@ -32,7 +33,8 @@
  * @since 4.4.0
  */
 @Beta
-public interface File extends Resource, VirtualFile, ModificationTracker, HasVcsStatus {
+public interface File
+    extends Resource, VirtualFile, ModificationTracker, HasVcsStatus, HasLocation {
```

Comment:
Extend `VirtualFile` with `HasLocation`. Because in nearest future, `File` interface will be deprecated and replaced with the new one universal resource interface.